src/tasks/AppleAppBuilder/AppleAppBuilder.cs

```diff
@@ -167,10 +166,76 @@ public string TargetOS
     /// </summary>
     public bool StripSymbolTable { get; set; }
 
+    /// <summary>
+    /// Bundles the application for NativeAOT runtime. Default runtime is Mono.
+    /// </summary>
+    public bool UseNativeAOTRuntime { get; set; }
+
+    public void ValidateRuntimeSelection()
+    {
+        if (UseNativeAOTRuntime)
+        {
+            if (!string.IsNullOrEmpty(MonoRuntimeHeaders))
+            {
+                Log.LogMessage(MessageImportance.High, $"Property \"{nameof(MonoRuntimeHeaders)}\" is not supported with NativeAOT runtime and will be ignored.");
+                MonoRuntimeHeaders = string.Empty;
+            }
+            if (!string.IsNullOrEmpty(MainLibraryFileName))
+            {
+                Log.LogMessage(MessageImportance.High, $"Property \"{nameof(MainLibraryFileName)}\" is not supported with NativeAOT runtime and will be ignored.");
+                MainLibraryFileName = string.Empty;
+            }
+            if (UseConsoleUITemplate)
+            {
+                Log.LogMessage(MessageImportance.High, $"Property \"{nameof(UseConsoleUITemplate)}\" is not supported with NativeAOT runtime and will be ignored.");
+                UseConsoleUITemplate = false;
+            }
+            if (ForceInterpreter)
+            {
+                Log.LogMessage(MessageImportance.High, $"Property \"{nameof(ForceInterpreter)}\" is not supported with NativeAOT runtime and will be ignored.");
+                ForceInterpreter = false;
+            }
+            if (ForceAOT)
+            {
+                Log.LogMessage(MessageImportance.High, $"Property \"{nameof(ForceAOT)}\" is not supported with NativeAOT runtime and will be ignored.");
+                ForceAOT = false;
+            }
+            if (!string.IsNullOrEmpty(RuntimeComponents))
+            {
+                Log.LogMessage(MessageImportance.High, $"Property \"{nameof(RuntimeComponents)}\" is not supported with NativeAOT runtime and will be ignored.");
+                RuntimeComponents = string.Empty;
+            }
+            if (!string.IsNullOrEmpty(DiagnosticPorts))
+            {
+                Log.LogMessage(MessageImportance.High, $"Property \"{nameof(DiagnosticPorts)}\" is not supported with NativeAOT runtime and will be ignored.");
+                DiagnosticPorts = string.Empty;
+            }
+            if (EnableRuntimeLogging)
+            {
+                Log.LogMessage(MessageImportance.High, $"Property \"{nameof(EnableRuntimeLogging)}\" is not supported with NativeAOT runtime and will be ignored.");
+                EnableRuntimeLogging = false;
+            }
+            if (EnableAppSandbox)
```

Comment:
this one should be independent of the runtime

Reply:
EnableAppSandbox enabled.